pkg/clientcert/cert_controller.go

```diff
@@ -183,25 +185,16 @@ func (c *clientCertificateController) sync(ctx context.Context, syncCtx factory.
 		return nil
 	}
 
-	// add additional data into client certificate secret
-	newSecretConfig := map[string][]byte{}
-	for k, v := range secret.Data {
-		newSecretConfig[k] = v
-	}
-	for k, v := range c.AdditonalSecretData {
-		newSecretConfig[k] = v
-	}
-	if !reflect.DeepEqual(newSecretConfig, secret.Data) {
-		secret.Data = newSecretConfig
-		if err := c.saveSecret(secret); err != nil {
-			return err
-		}
-	}
-
 	// create a csr to request new client certificate if
 	// a. there is no valid client certificate issued for the current cluster/agent
 	// b. client certificate exists and has less than a random percentage range from 20% to 25% of its life remaining
 	if c.hasValidClientCertificate(secret) {
+		// add additional data into client certificate secret
+		if err := c.appendAdditinalData(secret); err != nil {
+			return err
+		}
```

Comment:
since we save secret here https://github.com/open-cluster-management/registration/pull/130/files#diff-710e745dc0a42d985a34b2b2b79aaa8d5d559756ad0f5047f1c6922cbf8beb7dL179, why do we need to do  it again here?

Reply:
Here is to save the additional data, like agent name which might be missing and recovered from cert